specification/dartLangSpec.tex

```diff
@@ -15773,6 +15768,8 @@ \subsection{Yield-Each}
 Let $T_f$ be the element type of $f$
 (\ref{functions}),
 and let $T$ be the static type of $e$.
+It is a compile-time error if $T$ may not be assigned to
+the declared return type of $f$.
 If $f$ is a generator and
 $T$ implements \code{Iterable<$U$>} or \code{Stream<$U$>} for some $U$
 (\ref{interfaceSuperinterfaces})
```

Comment:
Also need to say that:

* if $f$ is a synchronous generator function, it's a compile-time error if $T$ is not assignable to \code{Iterable<$T_f$>}, and
* otherwise $f$ is a synchronous generator function, and it's a compile-time error if $T$ is not assignable to \code{Stream<$T_f$>}.

Otherwise you can statically get away with:
```dart
Object foo() sync* {
  yield* Stream.fromIterable([1, 2]);
}
```
without getting caught.

Reply:
Right! Done.